esp.c

```diff
@@ -31,40 +31,93 @@
 #endif
 
 
-
 #ifdef HAS_ESP_IDF_5
 
+static SemaphoreHandle_t show_mutex = NULL;
+
 void espShow(uint8_t pin, uint8_t *pixels, uint32_t numBytes, boolean is800KHz) {
-  rmt_data_t led_data[numBytes * 8];
+  // Note: Because rmtPin is shared between all instances, we will
+  //  end up releasing/initializing the RMT channels each time we
+  //  invoke on different pins. This is probably ok, just not
+  //  efficient. led_data is shared between all instances but will
+  //  be allocated with enough space for the largest instance; data
+  //  is not used beyond the mutex lock so this should be fine.
+
+#define SEMAPHORE_TIMEOUT_MS 50
+
+  static rmt_data_t *led_data = NULL;
+  static uint32_t led_data_size = 0;
+  static int rmtPin = -1;
+
+  if (show_mutex && xSemaphoreTake(show_mutex, SEMAPHORE_TIMEOUT_MS / portTICK_PERIOD_MS) == pdTRUE) {
```

Comment:
I like the idea of adding more feedback for users. I was trying to minimize the amount of code/data I added, to avoid bloating the size of the library.

Reply:
Yeah, I think in this library we can 100% assume Arduino-on-ESP-IDF. It uses the RMT library from arduino-esp32 which in turn depends on the RMT driver in ESP-IDF, and we are ourselves using FreeRTOS mutexes, so there's no reason to overthink things.

And by the time anyone's user code is running, and they're calling `show()`, I think all the regular logging methods should be just fine. I'd follow existing Adafruit convention for whether to use `log_e` or `ESP_LOGE`.